.github/workflows/ci.yml

```diff
@@ -0,0 +1,23 @@
+# This can be run manually like so:
+#
+#     gh workflow run ci.yml --ref <git_branch_name>
+#
+name: Continuous Integration
+on:
+  pull_request:
+  push:
+  workflow_dispatch:
+jobs:
+  build:
+    name: nix-build
+    runs-on: self-hosted
+    steps:
+
+    - name: Checkout repository to $GITHUB_WORKSPACE
+      uses: actions/checkout@v2.3.4
+
+    - run: nix develop --command "./bin/ci-format-check"
```

Comment:
This is a big misuse of `nix develop` IMO. We want to test `nix build`, `nix develop` isn't pure, however, it isn't worth fixing right now as I assume that @nixinator will be doing some hercules stuff anyway.